### PR TITLE
Fix #9476: Fix accidental note input bar and palette interaction

### LINKED ISSUE #9476: [MU4 Issue] When an accidental is selected on the score, clicking an accidental in the Note Input Bar does nothing
**Describe the bug**
When an accidental is selected on the score, I would expect intuitively that I can change that accidental by clicking a different accidental in the Note Input Bar, or in the Accidentals Palette. This is not the case; instead, nothing happens. I can only change the accidental by first selecting the note.

### PR DESCRIPTION
**Pull request by iwoithe**
Resolves: #9476

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/engraving/libmscore/accidental.cpp [hunk 1]
```diff
@@ -22,6 +22,10 @@
 
 #include "accidental.h"
 
+#include "editdata.h"
+#include "engravingitem.h"
+#include "engravingobject.h"
+#include "infrastructure/symbolfont.h"
```
Comment (cbjeukendrup) [inline src/engraving/libmscore/accidental.cpp:28]: These four #includes are unnecessary/undesired :)

File: src/engraving/libmscore/cmd.cpp [hunk 2]
```diff
@@ -1909,6 +1911,21 @@ void Score::toggleAccidental(AccidentalType at, const EditData& ed)
 
 void Score::changeAccidental(AccidentalType idx)
 {
+    if (selection().isAccentidentalsOnly()) {
```
Comment (cbjeukendrup) [inline src/engraving/libmscore/cmd.cpp:1914]: I think we can remove this "if" condition and return statement; we should allow a mixed selection. For example, if my selection is something like in the first measure of the following example, I would like to end up with something like the second measure:
Scherm­afbeelding 2022-12-02 om 23 26 46

File: src/engraving/libmscore/cmd.cpp [hunk 3]
```diff
@@ -27,6 +27,8 @@
 
 #include <assert.h>
 
+#include "engravingitem.h"
+#include "engravingobject.h"
```
Comment (cbjeukendrup) [inline src/engraving/libmscore/cmd.cpp:31]: These includes are also not necessary. They happen automatically by the other includes. 

(You're right that in general these "transitive includes" are not always great, but in the case of these particular headers, I think we should allow it.)

**Review by cbjeukendrup — COMMENTED** [review]
Thanks for this! 

One general point: we prefer if you _rebase_ the PR branch on top of master, instead of _merging_ master into your PR branch. Those merge commits inside PRs make the history difficult to understand...

File: src/engraving/libmscore/cmd.cpp (hunk 2 above)
Comment (iwoithe) [reply src/engraving/libmscore/cmd.cpp:1914]: Do I then remove the `isAccentidentalsOnly()` method (as it's not used anymore)?

Comment (cbjeukendrup) [reply src/engraving/libmscore/cmd.cpp:1914]: Yes, indeed.

File: src/engraving/libmscore/accidental.cpp [hunk 4]
```diff
@@ -559,6 +563,9 @@ EngravingItem* Accidental::drop(EditData& data)
 {
     EngravingItem* e = data.dropElement;
     switch (e->type()) {
+    case ElementType::ACCIDENTAL:
+        score()->changeAccidental(toAccidental(e)->accidentalType());
```
Comment (cbjeukendrup) [inline src/engraving/libmscore/accidental.cpp:567]: I think you need to change this to <code>score()->changeAccidental(<b>note()</b>, toAccidental(e)->accidentalType());</code> to fix the issue you asked about on Discord 🙂

**Review by RomanPudashkin — APPROVED** [review]

**Comment by iwoithe** [comment]
@RomanPudashkin Rebased and squashed. :slightly_smiling_face:

**Comment by zacjansheski** [comment]
Tested on Mac OS 11.6.5, Ubuntu 20.04.6, Windows 11
https://github.com/musescore/MuseScore/issues/9476 FIXED

**Pull request merged**